### DISCUSSION TITLE
Restic 0.14.0 Released

### DISCUSSION THREAD

**Discussion by CoordSpace** [djmaze/resticker#143] [General]
The upstream update I know many of us have been waiting for has finally dropped. Along with compression, there are a number of other options and features that users will probably want exposed for configuration in resticker. Here are the release highlights:

https://restic.net/blog/2022-08-25/restic-0.14.0-released/

------------------------------

This release contains a highly anticipated feature in restic - **compression**! To start using it, either [initialize a new repository](https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html#preparing-a-new-repository) (which defaults to repository version 2 and therefore supports compression) or [upgrade your current repository](https://restic.readthedocs.io/en/stable/045_working_with_repos.html#upgrading-the-repository-format-version). The [compression level](https://restic.readthedocs.io/en/stable/047_tuning_backup_parameters.html#compression) can be changed from the default auto (fast and efficient) to max or off using the --compression option (or the corresponding [environment variable](https://restic.readthedocs.io/en/stable/040_backup.html#environment-variables)). Please note that the new/upgraded repository can only be accessed by restic version 0.14 or higher.

Another feature that can be highly useful is the --unsafe-recover-no-free-space option to the prune command, which allows you to save the day when the repository storage has been completely saturated. Just make sure to [read the corresponding documentation](https://restic.readthedocs.io/en/stable/060_forget.html#recovering-from-no-free-space-errors) first!

Besides the above, here are some other notable fixes and improvements in this release:

* Ability to [tune the number of concurrent backend connections](https://restic.readthedocs.io/en/stable/047_tuning_backup_parameters.html#backend-connections), increasing backup performance.
* Ability to [increase pack size](https://restic.readthedocs.io/en/stable/047_tuning_backup_parameters.html#pack-size), increasing backup performance and using less storage resources.
* The check --read-data and prune commands no longer uses temporary files when reading from the repository, resulting in less disk I/O.
* The copy command’s performance has been greatly improved.
* The init and copy commands have a new --from-repo option, replacing the now deprecated --repo2 option.
* The backup command now uses around 30% less memory when backing up directories with hundreds of thousands (or more) files.

--------------------

It isn't recommended to upgrade your resticker-managed repos yet since it's still using the previous version but I did want to post the upgrade instructions since it will be a useful reference as 0.14.X support is patched into the project:

https://restic.readthedocs.io/en/stable/045_working_with_repos.html#upgrading-the-repository-format-version

Cheers and thanks for the wonderful software!

**Comment by djmaze** [comment]
Yes, very nice!

**Reply by Jimmy-SafeCash** [reply]
@djmaze How do we specify compression in your image? (I'm using docker compose if that makes a difference). Will I have to delete existing backups or will it apply it from this point onwards?

**Reply by djmaze** [reply]
@Jimmy-SafeCash I did not try it yet, but it seems if you either have a repository in format version 2 or you upgrade your existing repository (which might work using `docker compose exec restic migrate upgrade_repo_v2`), restic will from then on [use compression automatically](https://restic.readthedocs.io/en/stable/047_tuning_backup_parameters.html#compression).
